src/units/gender.hpp

```diff
@@ -19,6 +19,23 @@
 class t_string;
 class unit_gender;
 
+class gender_list {
```

Comment:
This seems really overcomplicated for no real reason...

Reply:
It's about as simple as you can make an iterator system, it seems.
I added it to get rid of the `from_int` lookup, and convert integer iteration to range-for.

Reply:
Yeah, uhh... it seems overcomplicated _because_ it's an iterator system for a set of _two fixed values_.